lightning/src/ln/channel.rs

```diff
@@ -372,8 +372,7 @@ pub(super) enum ChannelError {
 	Ignore(&'static str),
 	Close(&'static str),
 	CloseDelayBroadcast {
-		msg: &'static str,
-		update: Option<ChannelMonitor>
```

Comment:
Seems that this field wasn't needed, because the object is available at the point where this error is handled.

Reply:
Hmm, fine for now, I suppose, eventually it should be an opaque update object, but tht'll be after #387.

Reply:
OK

Reply:
Oops, right, so I missed that this is largely as an indication as to whether or not we need to do an update-add, less to provide it. Unless there's a reason you need to drop it for this commit, we should back this change out.

Reply:
Would it be OK to have it as a bool then?  otherwise, `ChannelError` gets a generic type param and that's a bunch of clutter in the rest of the file.

Reply:
Also noticing that it's only constructed in one place.

Reply:
Hmm, they're just internal types, having to template them in the return types of many Channel functions doesn't seem to bad. At a high level its pretty awkward to access anything in Channel after it returns a ChannelError::Close, since at that point, it is assumed that the Channel is in an invalid state. Its *probably* fine in nearly every case, but we should avoid it where possible.

Reply:
OK, the revert is done.  Let me know once the new commits are reviewed and I'll squash.